File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
import java.util.*;

class Solution {
    public int solution(int a, int b) {
        
        int m = Math.min(a,b);
        
        for (int i = m; i>=1; i--) {
            if (a%i==0 && b%i==0) {
                a /= i;
                b /= i;
                break;
            }
        }
        
//         b /= gcd;
        
        while (b%2==0) {
            b /= 2;
        }
        
        while (b%5==0) {
            b /= 5;
        }
        
        return (b == 1) ? 1 : 2;
    }
    
    // 유클리드 호제법
    private int getGcd(int a, int b) {
        if (b==0) {
            return a;
        }
        return getGcd(b, a%b);
    }
}

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    answer = 0
    
    m = min(a,b)
    
    for i in range(m, 0, -1):
        if a%i == 0 and b%i == 0:
            a = a//i
            b = b//i
    
    number = [2,5]
    
    if b == 1:
        return 1
    
    for n in number:
        while True:
            if b == 1:
                return 1
            if b%n == 0:
                b = b//n
                continue
            break
    
    return 2